### ISSUE TITLE
TypeError: search() got an unexpected keyword argument 'doc_type'

### ISSUE THREAD

**Issue by zhongxian93** [bitsensor/elastalert#105]
**Elasticsearch version: 7.0.1**

Hi all, i ran, after configuring my host and port to my elasticsearch 
```
docker run -p 3030:3030 -p 3333:3333 \
>     -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml \
>     -v `pwd`/config/elastalert-test.yaml:/opt/elastalert/config-test.yaml \
>     -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json \
>     -v `pwd`/rules:/opt/elastalert/rules \
>     -v `pwd`/rule_templates:/opt/elastalert/rule_templates \
>     --net="host" \
>     --name elastalert bitsensor/elastalert:latest

```

```
WARNING: Published ports are discarded when using host network mode

> @bitsensor/elastalert@2.0.1 start /opt/elastalert-server
> sh ./scripts/start.sh

09:41:30.245Z  INFO elastalert-server: Config:  No config.dev.json file was found in /opt/elastalert-server/config/config.dev.json.
09:41:30.248Z  INFO elastalert-server: Config:  Proceeding to look for normal config file.
09:41:30.253Z  INFO elastalert-server: Config:  A config file was found in /opt/elastalert-server/config/config.json. Using that config.
09:41:30.261Z  INFO elastalert-server: Router:  Listening for GET request on /.
09:41:30.261Z  INFO elastalert-server: Router:  Listening for GET request on /status.
09:41:30.262Z  INFO elastalert-server: Router:  Listening for GET request on /status/control/:action.
09:41:30.262Z  INFO elastalert-server: Router:  Listening for GET request on /status/errors.
09:41:30.262Z  INFO elastalert-server: Router:  Listening for GET request on /rules.
09:41:30.264Z  INFO elastalert-server: Router:  Listening for GET request on /rules/:id.
09:41:30.264Z  INFO elastalert-server: Router:  Listening for POST request on /rules/:id.
09:41:30.264Z  INFO elastalert-server: Router:  Listening for DELETE request on /rules/:id.
09:41:30.264Z  INFO elastalert-server: Router:  Listening for GET request on /templates.
09:41:30.264Z  INFO elastalert-server: Router:  Listening for GET request on /templates/:id.
09:41:30.264Z  INFO elastalert-server: Router:  Listening for POST request on /templates/:id.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for DELETE request on /templates/:id.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for POST request on /test.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for GET request on /config.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for POST request on /config.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for POST request on /download.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for GET request on /metadata/:type.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for GET request on /mapping/:index.
09:41:30.265Z  INFO elastalert-server: Router:  Listening for POST request on /search/:index.
09:41:30.268Z  INFO elastalert-server: ProcessController:  Starting ElastAlert
09:41:30.269Z  INFO elastalert-server: ProcessController:  Creating index
09:41:31.004Z  INFO elastalert-server:
    ProcessController:  Elastic Version:7
    Mapping used for string:{'type': 'keyword'}
    Index elastalert_status already exists. Skipping index creation.

09:41:31.004Z  INFO elastalert-server: ProcessController:  Index create exited with code 0
09:41:31.004Z  INFO elastalert-server: ProcessController:  Starting elastalert with arguments [none]
09:41:31.012Z  INFO elastalert-server: ProcessController:  Started Elastalert (PID: 47)
09:41:31.014Z  INFO elastalert-server: Server:  Server listening on port 3030
09:41:31.015Z  INFO elastalert-server: Server:  Websocket listening on port 3333
09:41:31.015Z  INFO elastalert-server: Server:  Server started
09:41:41.748Z ERROR elastalert-server:
    ProcessController:  Traceback (most recent call last):
      File "/usr/lib/python2.7/runpy.py", line 174, in _run_module_as_main

09:41:41.749Z ERROR elastalert-server:
    ProcessController:      "__main__", fname, loader, pkg_name)
      File "/usr/lib/python2.7/runpy.py", line 72, in _run_code
        exec code in run_globals
      File "/opt/elastalert/elastalert/elastalert.py", line 1929, in <module>
        sys.exit(main(sys.argv[1:]))
      File "/opt/elastalert/elastalert/elastalert.py", line 1925, in main
        client.start()
      File "/opt/elastalert/elastalert/elastalert.py", line 1106, in start

09:41:41.749Z ERROR elastalert-server:
    ProcessController:      self.run_all_rules()
      File "/opt/elastalert/elastalert/elastalert.py", line 1158, in run_all_rules

09:41:41.750Z ERROR elastalert-server:
    ProcessController:      self.send_pending_alerts()
      File "/opt/elastalert/elastalert/elastalert.py", line 1534, in send_pending_alerts
        pending_alerts = self.find_recent_pending_alerts(self.alert_time_limit)
      File "/opt/elastalert/elastalert/elastalert.py", line 1526, in find_recent_pending_alerts
        size=1000)
      File "/usr/lib/python2.7/site-packages/elasticsearch-7.0.1-py2.7.egg/elasticsearch/client/utils.py", line 84, in _wrapped
        return func(*args, params=params, **kwargs)
    TypeError: search() got an unexpected keyword argument 'doc_type'

09:41:41.779Z ERROR elastalert-server: ProcessController:  ElastAlert exited with code 1
09:41:41.779Z  INFO elastalert-server: Server:  Stopping server
09:41:41.779Z  INFO elastalert-server: ProcessController:  ElastAlert is not running
09:41:41.779Z  INFO elastalert-server: Server:  Server stopped. Bye!

```

i got the same error when i was trying it out the same elastalert version  on **elasticsearch version 6.7.0**  do anyone has any idea why this is occuring, as previously i found online the `doc_type` problem was due to doc_type depreciating in elastic search 7

Any help is appreciated, i have been stuck on this issue for 2 days already.
Thanks

**Comment by endarque** [comment]
I have same issue. I think the python /usr/lib/python2.7/site-packages/elasticsearch-7.0.1-py2.7.egg/ is problem. ES7 can't support 'doc_type'.

**Comment by HamidSajjadi** [comment]
> You can use this release by using bitsensor/elastalert:3.0.0-beta.0 in you Docker command.

had this same issue for the last 2 days. my problem was that Although I thought I was using version [3.0.0-beta](https://github.com/bitsensor/elastalert/releases/tag/3.0.0-beta.0) which is compatible with elastAlert v0.2.0b2 (which is compatible with elasticsearch 7),  was not using version 3.0.0-beta in my docker command and was copy-pasting it from the readme file. 

To be honest documentation could be a little better.

**Comment by zhongxian93** [comment]
hi i tried running 
```
docker run bitsensor/elastalert:3.0.0-beta.0 -p 3030:3030 -p 3333:3333 \
    -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml \
    -v `pwd`/config/elastalert-test.yaml:/opt/elastalert/config-test.yaml \
    -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json \
    -v `pwd`/rules:/opt/elastalert/rules \
    -v `pwd`/rule_templates:/opt/elastalert/rule_templates \
    --net="host" \
    --name elastalert bitsensor/elastalert:latest
```

after the image is downloaded i got an output `6.9.0` with no other logs appearing 
and the container stopped am i doing something wrong and is bitsensor/elastalert:3.0.0-beta.0  actually working?

**Comment by HamidSajjadi** [comment]
Try this

```
docker run -d -p 3030:3030 -p 3333:3333 \
    -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml \
    -v `pwd`/config/elastalert-test.yaml:/opt/elastalert/config-test.yaml \
    -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json \
    -v `pwd`/rules:/opt/elastalert/rules \
    -v `pwd`/rule_templates:/opt/elastalert/rule_templates \
    --net="host" \
    --name elastalert bitsensor/elastalert:3.0.0-beta.0
```

And I needed to delete indexes that elastalert had created on my elasticsearch so they are created again by this version.

> is bitsensor/elastalert:3.0.0-beta.0 actually working?

I'm not sure it is working or not, because I encountered #101 further ahead and still could not fix it. maybe the problem is not in elastalert:3.0.0-beta.0 and is something else I'm doing wrong. 
But at least the docker and rest API started.

**Comment by zhongxian93** [comment]
i just tried the command you suggested and the elastalert finally started, however the plugin fail to connect 
```
502 - Bad Gateway
Client request error: connect EHOSTUNREACH <myipaddress>:3030
```

i have even tried to delete the elastalert indexes and nothing changed, do you have any idea why or am i missing some steps?
______________________________________________________________________________________________________________
Update managed to get it to work after attaching `network` options, apparently i am supposed to run it in the same network as my docker elk. Here is the command for anyone who needs it, and removing the     --net="host" \ options helped too

```
	docker run -p 3030:3030 -p 3333:3333 --network="docker-elk_elk" \
    -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml \
    -v `pwd`/config/elastalert-test.yaml:/opt/elastalert/config-test.yaml \
    -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json \
    -v `pwd`/rules:/opt/elastalert/rules \
    -v `pwd`/rule_templates:/opt/elastalert/rule_templates \
    --name elastalert bitsensor/elastalert:3.0.0-beta.0
```
after running the new command i have to change
```
elastalert-kibana-plugin.serverHost: <myipaddress>
elastalert-kibana-plugin.serverPort: 3030

```
to
```
elastalert-kibana-plugin.serverHost: elastalert
elastalert-kibana-plugin.serverPort: 3030

```
for now i have encountered the #101 too, will update if managed to solve it

**Comment by zhongxian93** [comment]
I have managed to solved it thanks for all the help

**Comment by rahulsinghidrive** [comment]
@zhongxian93 can you share the steps to make it work. I am having the same issue and I am using:
- elasticsearch: 6.7.1
- elastalert: 2.0.1

**Comment by zhongxian93** [comment]
I am unclear what problem you are encountering. 
If you are encountering `TypeError: search() got an unexpected keyword argument 'doc_type'`
i managed to fix it  using elastalert: 3.0.0-beta.0 along with elastalert7.0.1 by running this command :
```
docker run -d -p 3030:3030 -p 3333:3333 \
    -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml \
    -v `pwd`/config/elastalert-test.yaml:/opt/elastalert/config-test.yaml \
    -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json \
    -v `pwd`/rules:/opt/elastalert/rules \
    -v `pwd`/rule_templates:/opt/elastalert/rule_templates \
    --net="host" \
    --name elastalert bitsensor/elastalert:3.0.0-beta.0
```
if your issue was 
```
502 - Bad Gateway
Client request error: connect EHOSTUNREACH <myipaddress>:3030
```
in my case i was hosing my elasticsearch another network hence elasalert is unable to reach it.
SO i have managed to fixed it by running the docker in the same network as my elastic search 
by using this command:
```
docker run -p 3030:3030 -p 3333:3333 --network="docker-elk_elk" \
    -v `pwd`/config/elastalert.yaml:/opt/elastalert/config.yaml \
    -v `pwd`/config/elastalert-test.yaml:/opt/elastalert/config-test.yaml \
    -v `pwd`/config/config.json:/opt/elastalert-server/config/config.json \
    -v `pwd`/rules:/opt/elastalert/rules \
    -v `pwd`/rule_templates:/opt/elastalert/rule_templates \
    --name elastalert bitsensor/elastalert:3.0.0-beta.0
```
